es/separable_natural_es.py

```diff
@@ -65,26 +78,31 @@ def optimize(func, mu, sigma,
         else:
             utility = fitness
 
-        # update parameter of search distribution via natural gradient descent in natural coordinates
-        mu += learning_rate_mu * sigma * np.dot(utility, s)
-        sigma *= np.exp(learning_rate_sigma / 2. * np.dot(utility, s ** 2 - 1))
-
         if record_history:
-            history_mu.append(mu.copy())
+            history_mu.append(mu_numpy.copy())
             history_sigma.append(sigma.copy())
             history_pop.append(z.copy())
             history_fitness.append(fitness.copy())
 
-        generation += 1
-
         # exit if max iterations reached
         if generation > max_iter or np.all(sigma < 1e-10):
             break
 
-    return {'mu': mu,
+        # update parameters of search distribution via natural
+        # gradient descent in natural coordinates
+
+        # set gradient and use optimizer to update mu
+        mu_torch.grad = torch.autograd.Variable(torch.DoubleTensor(-sigma * np.dot(utility, s)))
+        optimizer_mu.step()
+
+        # manually update sigma
+        sigma *= np.exp(learning_rate_sigma / 2. * np.dot(utility, s ** 2 - 1))
```

Comment:
Would it make sense here to update `log(sigma)` with the pytorch-optimizer? Or is that a bad idea due to numerical stability issues?